model/qti/CustomInteractionAsset/Extractor/TextReaderAssetExtractor.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ *
+ *
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiItem\model\qti\CustomInteractionAsset\Extractor;
+
+use oat\taoQtiItem\model\qti\CustomInteractionAsset\Extractor\Api\AssetExtractorInterface;
+use oat\taoQtiItem\model\qti\interaction\CustomInteraction;
+
+/**
+ * @author Kiryl Poyu <kyril.poyu@taotesting.com>
+ */
+class TextReaderAssetExtractor implements AssetExtractorInterface
+{
+    public const INTERACTION_IDENTIFIER = 'textReaderInteraction';
+    public const CONTENT_PREFIX = 'content-';
+
+    /**
+     * @inheritDoc
+     */
+    public function extract(CustomInteraction $interaction): array
+    {
+        $extractedAssets = [];
+        $contentProperties = array_filter($interaction->getProperties(), static function ($propertyKey) {
+            return strpos($propertyKey, self::CONTENT_PREFIX) !== false;
```

Comment:
Considering the `'content-'` value is a prefix, WDYT of checking if the key actually starts from it, not just contains it?
```suggestion
            return strpos($propertyKey, self::CONTENT_PREFIX) === 0;
```

Reply:
LGTM